clang/include/clang/Basic/DiagnosticSemaKinds.td

```diff
@@ -2809,10 +2813,13 @@ def note_non_literal_virtual_base : Note<"%select{struct|interface|class}0 "
   "with virtual base %plural{1:class|:classes}1 is not a literal type">;
 def note_constexpr_virtual_base_here : Note<"virtual base class declared here">;
 def err_constexpr_non_literal_return : Error<
-  "%select{constexpr|consteval}0 function's return type %1 is not a literal type">;
+  "%select{constexpr|consteval}0 function with non-literal return type %1 is a C++23 extension">;
```

Comment:
This should be reverted back to the original diagnostic wording; we no longer treat it as an extension.

Reply:
oops, still learning. Removed.